fsspec/implementations/tests/test_github.py

```diff
@@ -0,0 +1,41 @@
+import fsspec
+
+
+def test_github_open_small_file():
+    # test opening a small file <1 MB
+    with fsspec.open("github://mwaskom:seaborn-data@4e06bf0/penguins.csv") as f:
+        assert f.readline().startswith(b"species,island")
+
+
+def test_github_open_large_file():
+    # test opening a large file >1 MB
+    with fsspec.open("github://mwaskom:seaborn-data@83bfba7/brain_networks.csv") as f:
+        assert f.readline().startswith(b"network,1,1,2,2")
```

Comment:
Question: this will download the whole file in one shot, even though it's "large". Do you think we can generate a buffered file (like a HTTPFile), given that we expect the remote to accept HTTP range requests?

Reply:
This sounds like a great idea - thank you for mentioning this. I will rework this and then rewrite this test to use a range request to avoid fetching the whole file over the network.

Reply:
Do you have thoughts on if it would also be important to introduce other concepts from the http implementation such as support for async, reuse of an aiohttp ClientSession for the multiple calls to the GitHub API, etc.? Or would it be fine to just use HTTPFile and ensure that range requests are supported, but otherwise not reach for full feature parity with the http implementation in this first PR?

Reply:
I guess maybe my question may not make sense to ask if the relevant async support is already "enabled" just by virtue of the `asynchronous` kwarg on the HTTPFile constructor. I should probably take a closer look before asking this question 😅 .

Reply:
I did not think about the async-ness. We probably *don't* want this for github, since there is a fairly onerous API rate limit, even if you come with a developer token.

Reply:
In a751042 I tweaked `_open()` to return an HTTPFile wrapping the `download_url` in cases where the initial response from the GitHub API does not include the file content already. I believe this should enable streaming and range queries.